decompiler/analysis/analyze_inspect_method.cpp

```diff
@@ -920,7 +920,18 @@ std::string inspect_inspect_method(Function& inspect_method,
   result.flags = flags.flag;
   result.type_size = flags.size;
   result.type_method_count = flags.methods;
-  result.type_heap_base = flags.heap_base;
+
+  // Only set heap-base if it's different from the automatic one
+  // A child (or child of a child) of process ALWAYS has heap-base set.
+  if (flags.heap_base > 0) {
+    // TODO - how can i get this value in a non-hardcoded way
+    // - this is the size of a jak 2 "process"
```

Comment:
Maybe
```cpp
auto process_type = dts.ts->get_type_of_type<BasicType>("process");
auto auto_hb = (flags.size - process_type->size() + 0xf) & ~0xf;
```
I think that this requires us to have `process` in `all-types.gc` but we should have that already.

Reply:
If you want a per-game constant you can add something like this to `goal_constants.h`
```
constexpr PerGameVersion<int> GOAL_PROCESS_SIZE(0xwhatever, 0x80);
```
then `GOAL_PROCESS_SIZE[game_version]` will give you the right size.

Reply:
I remember trying this early on and i got an error, ill give it a shot and if it doesn't work ill use the second option.